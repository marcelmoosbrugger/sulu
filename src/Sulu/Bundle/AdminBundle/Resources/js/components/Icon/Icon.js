// @flow
import 'font-awesome/css/font-awesome.min.css';
import React from 'react';
import classNames from 'classnames';

type Props = {
    className?: string,
    onClick?: () => void,
    name: string,
};

export default class Icon extends React.PureComponent<Props> {
    render() {
        const {className, name, onClick} = this.props;
        const classes = classNames(className, 'fa', 'fa-' + name);

        return (
            <span className={classes} aria-hidden={true} onClick={onClick} />
        );
    }
}
