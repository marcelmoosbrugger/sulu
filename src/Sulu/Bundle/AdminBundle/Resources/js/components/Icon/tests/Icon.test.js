/* eslint-disable flowtype/require-valid-file-annotation */
import Icon from '../Icon';
import React from 'react';
import {render} from 'enzyme';

test('Icon should render', () => {
    expect(render(<Icon name="save" />)).toMatchSnapshot();
});

test('Icon should render with class names', () => {
    expect(render(<Icon className="test" name="edit" />)).toMatchSnapshot();
});
