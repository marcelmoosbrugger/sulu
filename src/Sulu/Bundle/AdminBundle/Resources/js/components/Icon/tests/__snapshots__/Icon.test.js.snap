// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render 1`] = `
<span
  aria-hidden="true"
  class="fa fa-save"
/>
`;

exports[`Icon should render with class names 1`] = `
<span
  aria-hidden="true"
  class="test fa fa-edit"
/>
`;
